core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -362,15 +391,128 @@ void writeUTF8Field(int fieldNumber, String value) throws IOException {
       abstract void writeBytes(byte[] value, int offset, int length) throws IOException;
 
       abstract void writeBytes(ByteBuffer value) throws IOException;
+
+      Encoder subEncoder(int number) throws IOException {
+         try (RandomAccessOutputStream raos = new RandomAccessOutputStreamImpl()) {
+            return new WrappedEncoder(new OutputStreamRandomAccessEncoder(raos)) {
+               @Override
+               void close() throws IOException {
+                  Encoder.this.writeBytes(number, raos.getByteBuffer());
+                  super.close();
+               }
+            };
+         }
+      }
+   }
```

Comment:
I'm confused about how this is supposed to work. When the `subEncoder` returns, the `close` method will be invoked in `RandomAccessOutputStream`. Should `raos.close` be invoked inside the `WrappedEncoder.close` method? 
Or just remove the try-with-resources.

Reply:
Honestly, this was an intermediate step and abuses the fact that all impls of RandomAccessOutputStream don't do anything on close. I can fix this commit. If you notice the last commit for this doesn't even invoke close anymore.